VideoLocker/src/main/java/org/edx/mobile/task/Task.java

```diff
@@ -22,32 +27,50 @@
 
     private ProgressBar progressBar;
 
-    private WeakReference<TaskProcessCallback> taskProcessCallback;
+    @Nullable
+    private WeakReference<TaskProgressCallback> progressCallback;
+    @Nullable
+    private WeakReference<TaskMessageCallback> messageCallback;
 
     protected final Handler handler = new Handler();
     protected final Logger logger = new Logger(getClass().getName());
 
     @Inject
     protected IEdxEnvironment environment;
-    
+
     public Task(Context context) {
         super(context);
 
-        if ( context instanceof TaskProcessCallback ){
+        if (context instanceof TaskProcessCallback) {
```

Comment:
This is a pretty hacking approach, but I guess we can let it be until we switch to an alternative solution. Same with the automatic weak referencing.